File: hw5/task4.py
from math import e, pow, pi

def trapezoidRule(f, a, b, n):
  h = (b - a) / n
  sum = 0.5 * (f(a) + f(b))
  for i in range(1, n):
    sum += f(a + i * h)
  return sum * h

tests = [2,4,8,16, 100000]

def f(x):
  return pow(e, -x * x)

for test in tests:
  print(f"n={test}; approx={trapezoidRule(f, 0, pi / 4, test)}")

File: hw6/task4.c
#include <stdio.h>
#include <math.h>

double factorial(int n)
{
  if (n == 0)
  {
    return 1;
  }
  return n * factorial(n - 1);
}

int main()
{
  double sum = 0;
  int n = 1000;
  for (int i = n; i >= 0; i--)
  {
    sum += 1 / factorial(i);
  }
  printf("e = %.16f\n", sum);
  return 0;
}
